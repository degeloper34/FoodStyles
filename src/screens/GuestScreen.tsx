import {LinearGradient} from "expo-linear-gradient";
import {Image, Pressable, StyleSheet, View} from "react-native";
import {RootStackScreenProps} from "../../types";
import {Colors, Spacing, TextStyles} from "../styles";
import {CustomText} from "../components/atoms";
import {AntDesign, FontAwesome} from "@expo/vector-icons";

export default function GuestScreen({
  navigation,
}: RootStackScreenProps<"GuestScreen">) {
  const onPressSignUpWithEmail = () => {
    navigation.navigate("SignUpScreen");
  };

  const onPressLogIn = () => {
    navigation.navigate("LoginScreen");
  };

  const {
    container,
    viewHeader,
    viewLogo,
    imgLogo,
    viewFoodStyles,
    viewWelcomeText,
    viewAuthOptions,
    iconSocial,
    btnLogIn,
    btnSignIn,
    viewTerms,
  } = styles;

  return (
    <LinearGradient
      colors={["rgb(250,119,69)", "rgb(243, 196, 66)"]}
      style={container}
    >
      <View style={viewHeader} />

      <View style={viewLogo}>
        <Image
          source={require("../assets/images/logo.png")}
          style={imgLogo}
          resizeMode={"contain"}
        />
        <View style={viewFoodStyles}>
          <CustomText
            style={TextStyles.TEXT_STYLE_8}
            text={"FoodStyles"}
            type={"bold"}
          />
        </View>

        <View style={viewWelcomeText}>
          <CustomText
            style={TextStyles.TEXT_STYLE_3}
            text={"Sign in to be able to save your\rpreferences and settings."}
            type={"regular"}
          />
        </View>
      </View>

      <View style={viewAuthOptions}>
        <Pressable style={btnSignIn}>
          <FontAwesome
            name="apple"
            size={20}
            color={Colors.GREYISH_BROWN}
            style={iconSocial}
          />
          <CustomText
            style={TextStyles.TEXT_STYLE_2}
            text={"Sign in with Apple"}
            type={"semi-bold"}
          />
        </Pressable>

        <Pressable style={btnSignIn}>
          <FontAwesome
            name="facebook-official"
            size={20}
            color="#3b5998"
            style={iconSocial}
          />
          <CustomText
            style={TextStyles.TEXT_STYLE_2}
            text={"Sign in with Facebook"}
            type={"semi-bold"}
          />
        </Pressable>

        <Pressable style={btnSignIn}>
          <AntDesign
            name="google"
            size={20}
            color="#4285f4"
            style={iconSocial}
          />
          <CustomText
            style={TextStyles.TEXT_STYLE_2}
            text={"Sign in with Google"}
            type={"semi-bold"}
          />
        </Pressable>

        <Pressable style={btnSignIn} onPress={onPressSignUpWithEmail}>
          <CustomText
            style={TextStyles.TEXT_STYLE_2}
            text={"Sign up with Email"}
            type={"semi-bold"}
          />
        </Pressable>

        <Pressable style={btnLogIn} onPress={onPressLogIn}>
          <CustomText
            style={TextStyles.TEXT_STYLE}
            text={"Log in with Email"}
            type={"semi-bold"}
          />
        </Pressable>
      </View>

      <View style={viewTerms}>
        <CustomText
          style={TextStyles.TEXT_STYLE_4}
          text={
            "By signing in you accept the\rGeneral Terms and Privacy Policy"
          }
          type={"semi-bold"}
        />
      </View>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  viewHeader: {
    flex: 1,
  },
  viewLogo: {
    flex: 2,
    alignItems: "center",
  },
  imgLogo: {
    width: 71,
    height: 84,
  },
  viewFoodStyles: {flex: 1, justifyContent: "center"},
  viewWelcomeText: {
    flex: 2,
    justifyContent: "center",
  },
  viewAuthOptions: {
    flex: 3,
    paddingHorizontal: 70,
  },
  iconSocial: {
    marginRight: Spacing.s,
  },
  btnLogIn: {
    marginTop: Spacing.l,
    alignItems: "center",
  },
  btnSignIn: {
    borderRadius: 100,
    backgroundColor: Colors.WHITE,
    shadowColor: Colors.GREYISH_BROWN_30_20,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 7,
    shadowOpacity: 1,
    alignItems: "center",
    justifyContent: "center",
    padding: Spacing.s,
    marginTop: Spacing.m,
    flexDirection: "row",
  },
  viewTerms: {
    flex: 1,
    justifyContent: "center",
  },
});
