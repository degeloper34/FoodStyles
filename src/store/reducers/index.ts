import {combineReducers} from "redux";

import memberReducers from "./memberReducers";

export const rootReducer = combineReducers({
  memberReducers,
});
