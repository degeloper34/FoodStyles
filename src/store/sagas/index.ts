import {all} from "redux-saga/effects";

import {memberWatcher} from "./memberSagas";

export default function* rootSaga() {
  yield all([memberWatcher()]);
}
