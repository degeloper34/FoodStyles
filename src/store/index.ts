export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

import {legacy_createStore as createStore} from "redux";
import {rootReducer} from "./reducers";

export const store = createStore(rootReducer);
