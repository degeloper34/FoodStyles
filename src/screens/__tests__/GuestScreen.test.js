import renderer from "react-test-renderer";
import React from "react";
import {Provider} from "react-redux";
import {store} from "../../store";
import GuestScreen from "../GuestScreen";

describe("GuestScreen Tests", () => {
  test("GuestScreen | Render without issues", () => {
    const screen = renderer
      .create(
        <Provider store={store}>
          <GuestScreen />
        </Provider>
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
