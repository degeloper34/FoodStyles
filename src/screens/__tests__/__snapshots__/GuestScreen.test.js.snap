// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestScreen Tests GuestScreen | Render without issues 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4294604613,
        4294165570,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 84,
          "width": 71,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "bold",
              "letterSpacing": 0,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-bold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        FoodStyles
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 18,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "letterSpacing": 0,
              "lineHeight": 22,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-regular",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Sign in to be able to save your
preferences and settings.
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 3,
        "paddingHorizontal": 70,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "padding": 12,
          "shadowColor": "rgba(111, 93, 80, 0.2)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#434343",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": 0,
              "lineHeight": 18,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Sign in with Apple
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "padding": 12,
          "shadowColor": "rgba(111, 93, 80, 0.2)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#434343",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": 0,
              "lineHeight": 18,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Sign in with Facebook
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "padding": 12,
          "shadowColor": "rgba(111, 93, 80, 0.2)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#434343",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": 0,
              "lineHeight": 18,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Sign in with Google
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 100,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "padding": 12,
          "shadowColor": "rgba(111, 93, 80, 0.2)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#434343",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": 0,
              "lineHeight": 18,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Sign up with Email
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": 0,
            },
            Object {
              "fontFamily": "proxima-semibold",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Log in with Email
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 13,
            "fontStyle": "normal",
            "fontWeight": "600",
            "letterSpacing": 0,
            "textAlign": "center",
          },
          Object {
            "fontFamily": "proxima-semibold",
            "textDecorationLine": "none",
          },
        ]
      }
    >
      By signing in you accept the
General Terms and Privacy Policy
    </Text>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
